informer/src/handler/connection.rs

```diff
@@ -0,0 +1,70 @@
+// Copyright 2020 Kodebox, Inc.
+// This file is part of CodeChain.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+use crate::{Handler, Sink, PubSubHandler, Session, MetaIoHandler, Params, Error, ErrorCode, SubscriptionId, Subscriber, Value, PubSubMetadata};
+use jsonrpc_core::{futures, BoxFuture};
+use futures::Future;
+use std::sync::Arc;
+use std::{thread, time};
+
+
+pub struct Connection{
+    pub status: bool,
+    pub intrested_event: Vec<String>,
+    pub sink: Sink,
+}
+
+impl Connection{
+    pub fn new(status: bool, events: String , sink: Sink) -> Self {
+
+        Self{
+            status: status,
+            intrested_event: vec![events.clone()],
+            sink: sink.clone(),
+        }
+    }
+    pub fn event_subscription <T> (&mut self, mut io : PubSubHandler<T>) {
+
+        io.add_subscription( "Your event subscription begins", ("register", move |params: Params, _, subscriber: Subscriber| {
+            let paramsclone = params.clone();
+            let unwraped_params: Result<String, Error> = params.parse();
+            let eventname=  unwraped_params.unwrap();
+
+            if paramsclone == Params::None {
+                subscriber
+                    .reject(Error{
+                        code: ErrorCode::ParseError,
+                        message: "Invalid parameters. Subscription rejected.".into(),
+                        data: None,
+                    })
+                    .unwrap();
+                return;
+            }
+
+            thread::spawn(move || {
```

Comment:
Why do we need a new thread here?

Reply:
This is for my experience of how we can manage a thread. do not mind it for now.